src/OpenTelemetry/Metrics/MeterProviderExtensions.cs

```diff
@@ -48,13 +48,13 @@ public static bool ForceFlush(this MeterProvider provider, int timeoutMillisecon
                 throw new ArgumentNullException(nameof(provider));
             }
 
-            if (provider is MeterProviderSdk meterProviderSdk)
+            if (timeoutMilliseconds < 0 && timeoutMilliseconds != Timeout.Infinite)
             {
-                if (timeoutMilliseconds < 0 && timeoutMilliseconds != Timeout.Infinite)
-                {
-                    throw new ArgumentOutOfRangeException(nameof(timeoutMilliseconds), timeoutMilliseconds, "timeoutMilliseconds should be non-negative.");
-                }
+                throw new ArgumentOutOfRangeException(nameof(timeoutMilliseconds), timeoutMilliseconds, "timeoutMilliseconds should be non-negative.");
```

Comment:
message doesn't include the case where `timeoutMilliseconds != Timeout.Infinite` append " and not infinite" or something

Reply:
How about appending `or Timeout.Infinite`?

Reply:
Fixed.